bondarilya/second/task1.py

```diff
@@ -0,0 +1,31 @@
+"""
+Вхідні дані: текст довільної довжини, який може містити літери латинського алфавіту, пробіли та розділові знаки (,.:;!?-);
+Результат роботи: список слів (у нижньому регістрі), які зустрічаються в тексті найменшу кількість раз.
+Слова, записані через дефіс, вважати двома словами (наприклад, "hand-made"). Слова у різних відмінках, числах та з іншими перетвореннями (наприклад, "page" та "pages") вважаються різними словами. Регістр слів -- навпаки, не має значення: слова "page" та "Page" вважаються 1 словом.
+
+Якщо слів, які зустрічаються найменше, декілька -- вивести їх в алфавітному порядку.
+Приклад: 'Hello,Hello, my dear!'
+Результат: [my,dear]
+
+"""
+import re
+re_str = re.compile("^\w*\,\.\:\;\!\?\-+$")
+
+
+a = input("Введіть текст : ")
+re_res = re.findall(re_str,a)
+print(re_res)
+
+
+"""b = a.split(" ")
+print(b)
+res = []
+for i in b:
+    if i.isdigit():
+        i = "1"
+        res += i
+    else:
+        res += i
```

Comment:
If you want to count how many times this word is in your line, try using dictionaries:
```
words = {:}
    for word in re_res:
         if word in words.keys():
               words[word] = words[word]+1
         else:
               words.update({word : 1})
```

Afterwards you can find the minimum amount of times word can be in your line. For example you can write:
```
minimum = 0
for times in words.values()
    if minimum == 0:
        minimum = times
   else:
        if minimum > times:
             minimum = times
```

Once you found that number, you are ready to create your final list.
```
my_words = []
for word, times in words.items():
     if times == minimum:
          my_words.append(word)
print(my_words)
```